### PR TITLE
Add Springer Journal Article template

### PR DESCRIPTION
**Pull request by strakaps**
This adds an `rticles::springer_article()` template based on the [Springer LaTeX template](https://www.springer.com/gp/livingreviews/latex-templates).

### CODE REVIEW

**Comment by jjallaire** [comment]
@yihui Could you review and merge this one?

**Comment by yihui** [comment]
Okay, I will.

File: inst/rmarkdown/templates/springer_article/skeleton/bibliography.bib [hunk 1]
```diff
@@ -0,0 +1,233 @@
+%% This BibTeX bibliography file was created using BibDesk.
+%% http://bibdesk.sourceforge.net/
```
Comment (yihui) [inline inst/rmarkdown/templates/springer_article/skeleton/bibliography.bib]: Could you only keep the bib entries that are actually used in the example article?

File: inst/rmarkdown/templates/asa_article/resources/template.tex [hunk 2]
```diff
@@ -24,8 +24,13 @@
 %% load any required packages here
 $preamble$
 
+$if(highlighting-macros)$
+$highlighting-macros$
+$endif$
+
```
Comment (yihui) [inline inst/rmarkdown/templates/asa_article/resources/template.tex]: Next time it will be better if you could separate this into a different pull request since it is not related to the new Springer format (i.e. [one thing per PR](https://yihui.name/en/2018/02/bite-sized-pull-requests/)).

File: inst/rmarkdown/templates/springer_article/skeleton/bibliography.bib [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
...ZlcnNpb25YJG9iamVjdHNZJGFyY2hpdmVyVCR0b3ASAAGGoKgHCBMUFRYaIVUkbnVsbNMJCgsMDxJXTlMua2V5c1pOUy5vYmplY3RzViRjbGFzc6INDoACgAOiEBGABIAFgAdccmVsYXRpdmVQYXRoWWFsaWFzRGF0YV8QMi4uLy4uL1BhcGVyLU1NIGNoYXB0ZXIvR2FseWFyZHQtY2hhcHRlci03LjguMTMucGRm0hcLGBlXTlMuZGF0YU8RAeIAAAAAAeIAAgAADE1hY2ludG9zaCBIRAAAAAAAAAAAAAAAAAAAAMm3i1VIKwAAAJ3/thtHYWx5YXJkdC1jaGFwdGVyLTcuOC4xMy5wZGYAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAu7Y5zgBejQAAAAAAAAAAAAIAAgAACSAAAAAAAAAAAAAAAAAAAAAQUGFwZXItTU0gY2hhcHRlcgAQAAgAAMm3w5UAAAARAAgAAM4Als0AAAABABAAnf+2ABefWwANUdgAAJg6AAIAVU1hY2ludG9zaCBIRDpVc2VyczoAYWdhbHlhcmR0OgBEcm9wYm94OgBQYXBlci1NTSBjaGFwdGVyOgBHYWx5YXJkdC1jaGFwdGVyLTcuOC4xMy5wZGYAAA4AOAAbAEcAYQBsAHkAYQByAGQAdAAtAGMAaABhAHAAdABlAHIALQA3AC4AOAAuADEAMwAuAHAAZABmAA8AGgAMAE0AYQBjAGkAbgB0AG8AcwBoACAASABEABIARFVzZXJzL2FnYWx5YXJkdC9Ecm9wYm94L1BhcGVyLU1NIGNoYXB0ZXIvR2FseWFyZHQtY2hhcHRlci03LjguMTMucGRmABMAAS8AABUAAgAQ//8AAIAG0hscHR5aJGNsYXNzbmFtZVgkY2xhc3Nlc11OU011dGFibGVEYXRhox0fIFZOU0RhdGFYTlNPYmplY3TSGxwiI1xOU0RpY3Rpb25hcnmiIiBfEA9OU0tleWVkQXJjaGl2ZXLRJidUcm9vdIABAAgAEQAaACMALQAyADcAQABGAE0AVQBgAGcAagBsAG4AcQBzAHUAdwCEAI4AwwDIANACtgK4Ar0CyALRAt8C4wLqAvMC+AMFAwgDGgMdAyIAAAAAAAACAQAAAAAAAAAoAAAAAAAAAAAAAAAAAAADJA==}}
```
Comment (yihui) [inline inst/rmarkdown/templates/springer_article/skeleton/bibliography.bib]: I'm not familiar with these `Bdsk-File-1` fields. Are they absolutely necessary? If not, I hope they could be removed.

File: R/springer_article.R [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+#' @return R Markdown output format to pass to \code{\link[rmarkdown:render]{render}}
+#' @examples
+#' \dontrun{
+#' rmarkdown::draft("MyArticle.Rmd", template = "springer_article", package = "rticles")
+#' }
+#'
+#' @export
+springer_article <- function(..., keep_tex = TRUE, citation_package = 'none'){
+  template <- find_resource("springer_article", "template.tex")
+  fmt <- inherit_pdf_document(template = template,
```
Comment (yihui) [inline R/springer_article.R]: I don't think it is necessary to create the variable `fmt` here. I mean the assignment is unnecessary.

File: inst/rmarkdown/templates/asa_article/resources/template.tex (hunk 2 above)
Comment (strakaps) [reply inst/rmarkdown/templates/asa_article/resources/template.tex]: Right! (Didn't know it's not related.)

File: inst/rmarkdown/templates/springer_article/skeleton/bibliography.bib (hunk 1 above)
Comment (strakaps) [reply inst/rmarkdown/templates/springer_article/skeleton/bibliography.bib]: done

**Comment by strakaps** [comment]
Thanks for the quick review @yihui , I made the changes
